src/components/AskQuestion/AskQuestion.vue

```diff
@@ -169,35 +178,39 @@ export default {
       // todo: check min-length of title/description?
       // afterwards delete this.title, this.description. Later on have to check all the attributes.
       let validation = this.validateQuestion();
-      if (validation && sessionStorage.getItem("userName")!= null) {
-      let fullDate = new Date();
-      let month = fullDate.getMonth() + 1;
-      let day = fullDate.getDate();
-      let year = fullDate.getFullYear();
-      this.questionCreated_at = `${day}.${month}.${year}`;
-      this.questionAuthor = sessionStorage.getItem("userName");
-      const questionToList = {
-        questionTitle: this.currentQuestion.questionTitle,
-        questionDescription: this.currentQuestion.questionDescription,
-        questionCategory: this.currentQuestion.questionCategory,
-        questionIsDone: this.currentQuestion.questionIsDone,
-        questionCreated_at: this.questionCreated_at,
-        questionAuthor: this.questionAuthor,
-        questionUpvotes: this.currentQuestion.questionUpvotes,
-        usersVotedQuestion: this.currentQuestion.usersVotedQuestion,
-      };
-        this.questionArray.push(questionToList);
-      // creates database entry with given questionToList
-      DataService.create(questionToList)
-        .then(() => {})
-        .catch((e) => {
-          console.error(e);
-        });
-        this.currentQuestion.questionTitle = "";
-        this.currentQuestion.questionDescription = "";
+      if (validation) {
+        if (sessionStorage.getItem("userName")!= null){
```

Comment:
Please check if the user logged in at init of the view and then display the form or not.
Access to the Form should be the first check and show/hide the form.
Validation of the form IF the form data is valid should be done separate here and should not be mixed with the user state.